2019Labs/CustomSecurityContent/documentation/lab3_oval.adoc

```diff
@@ -0,0 +1,487 @@
+    </criteria>
+  </definition>
+
+....
+
+Here, we can see that there are two tests - one checks for `TMOUT` setting in the `/etc/profile` file, the other one checks all files in `/etc/profile.d/` that have the `sh` file extension.
+The check evaluates as passed if any of the tests pass, i.e. if the `TMOUT` assignment is set in either of those files.
+
+The `/etc/profile` test and it's dependencies look like this:
+
+....
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="TMOUT in /etc/profile" id="test_etc_profile_tmout" version="1">                                                                    
+    <ind:object object_ref="object_etc_profile_tmout" />
+    <ind:state state_ref="state_etc_profile_tmout" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_etc_profile_tmout" version="1">
+    <ind:filepath>/etc/profile</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*TMOUT[\s]*=[\s]*(.*)[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_state id="state_etc_profile_tmout" version="1">
+    <ind:subexpression datatype="int" operation="greater than or equal" var_check="all" var_ref="var_accounts_tmout" />                                                                                           
+  </ind:textfilecontent54_state>
+
+  <external_variable comment="external variable for TMOUT" datatype="int" id="var_accounts_tmout" version="1" />
+
+....
+
+There are two regular expression that check for `TMOUT=...` in the `shared.xml` file, one for the `profile` test, and one for the `profile.d/*.sh` test.
+However, they are the same, and the alternative form of the assignment `export TMOUT=...` isn't handled in either of them.
+
+Moreover, there is the `equals` operation used to perform the match.
+As stated in the previous section, this looks wrong, as longer timeouts are less secure.
+
+
+== Tests Hans-on
+
+The `ComplianceAsCode` project is equipped with test suite that is very useful when defining what scenarios the check and the remediation are supposed to handle.
+The test suite sets up a system to a certain state, and runs the scan.
+Depending on the state of the system, the scanner is supposed to either pass, or fail.
+If the scanner fails, we execute the remediation, and the subsequent scan should pass.
+
+Let's go to the project root, and execute the test:
+`./build_product fedora` will build the datastream that contains the OVAL check that we are going to test.
+
+We will test the `accounts_tmout` rule included in the `ospp` profile of the RHEL8 datastream.
+We have to run the test suite as a superuser, as it involves spinning up a container that exposes an SSH port.
+We will do so by invoking `sudo python3 tests/test_suite.py rule --container ssg_test_suite --datastream build/ssg-fedora-ds.xml accounts_tmout`
```

Comment:
Small nitpick: I usually prefer to have long  commands as a separate block so they are visually separated from the rest of the text.